sktime/forecasting/rbf_forecaster.py

```diff
@@ -0,0 +1,523 @@
+
+        self.centers = nn.Parameter(centers, requires_grad=True)
+
+        valid_rbf_types = {"gaussian", "multiquadric", "inverse_multiquadric"}
+        if self.rbf_type not in valid_rbf_types:
+            raise ValueError(
+                f"rbf_type must be one of {valid_rbf_types}, got {self.rbf_type}"
+            )
+
+    def forward(self, x):
+        """Apply the RBF transformation to the input tensor.
+
+        Parameters
+        ----------
+        x : torch.Tensor
+            Input data tensor.
+
+        Returns
+        -------
+        torch.Tensor
+            Transformed tensor with RBF features.
+        """
+        if x.dim() == 1:
+            x = x.unsqueeze(0)
+
+        diff = x.unsqueeze(1) - self.centers.unsqueeze(0)
+
+        distances_squared = torch.sum(diff**2, dim=-1)
+
+        if self.rbf_type == "gaussian":
+            return torch.exp(-self.gamma * distances_squared)
+        elif self.rbf_type == "multiquadric":
+            return torch.sqrt(1 + self.gamma * distances_squared)
+        else:  # inverse_multiquadric
+            return 1 / torch.sqrt(1 + self.gamma * distances_squared)
+
+
+class RBFNetwork(nn.Module):
+    r"""Neural network with an RBF layer followed by fully connected layers.
+
+    This model is designed to use RBF-transformed features as input for a series
+    of linear transformations, enabling effective learning from non-linear
+    representations.
+
+    Parameters
+    ----------
+    input_size : int
+        Number of input features
+    hidden_size : int
+        Number of units in the RBF layer.
+    output_size : int
+        Number of output features for the network.
+    centers : torch.Tensor, optional (default=None)
+        Centers points for the RBF layer
+    gamma : float, optional (default=1.0)
+        Scaling factor controlling the spread of the RBF layer.
+    rbf_type : {"gaussian", "multiquadric", "inverse_multiquadric"},
+        optional (default="gaussian")
+        The type of RBF kernel to apply.
+        - "gaussian": :math:`\\exp(-\\gamma (t - c)^2)`
+        - "multiquadric": :math:`\\sqrt{1 + \\gamma (t - c)^2}`
+        - "inverse_multiquadric": :math:`\frac{1}{\\sqrt{1 + \\gamma (t - c)^2}}`
+
+    linear_layers : list of int, optional (default=[64, 32])
+        Sizes of linear layers following the RBF layer
+    """
+
+    def __init__(
+        self,
+        input_size,
+        hidden_size,
+        output_size,
+        centers=None,
+        gamma=1.0,
+        rbf_type="gaussian",
+        linear_layers=[64, 32],
+    ):
+        super().__init__()
+
+        self.rbf_layer = RBFLayer(
+            in_features=input_size,
+            out_features=hidden_size,
+            centers=centers,
+            gamma=gamma,
+            rbf_type=rbf_type,
+        )
+
+        layers = []
+        prev_size = hidden_size
+
+        for size in linear_layers:
+            layers.extend([nn.Linear(prev_size, size), nn.ReLU(), nn.Dropout(0.1)])
```

Comment:
You might make the activation and the dropout configurable via hyperparams. So the user can easily exchange ReLU or dropout